### DISCUSSION TITLE
Poor job with numbered lists

### DISCUSSION THREAD

**Discussion by ilcaa72** [explosion/spaCy#8436] [Help: Coding & Implementations] [answered]
i tried this on the spacy website, and it doesnt do a good job with numbered lists.  any solution to this issue?

```
purpose for which Django was created.
In his talk “Death to Wireframes, Long Live Rapid Prototyping,” Bermon Painter stated
that there are five steps to the rapid prototyping process:
1. Observe and analyze. This is the part of the process where you figure out your end-user goals. It is all about brainstorming with your teammates about what you want to build (e.g., Who are your users? What will each user want from your product?).
2. Build. Using HTML, CSS, and JavaScript, the team now works together to create a minimum viable product (MVP) using only gray box samples and simple layouts.
3. Ship. Create a seamless way for every single person who touches the code on your team to deploy and ship code for viewing and/or testing purposes.
```

### Result

```
that there are five steps to the rapid prototyping process:
1.
Observe and analyze.
This is the part of the process where you figure out your end-user goals.
It is all about brainstorming with your teammates about what you want to build (e.g., Who are your users?
What will each user want from your product?).


2.
Build.
Using HTML, CSS, and JavaScript, the team now works together to create a minimum viable product (MVP) using only gray box samples and simple layouts.


3.
Ship.
Create a seamless way for every single person who touches the code on your team to deploy and ship code for viewing and/or testing purposes.

```

**Comment by polm** [comment, marked as answer]
Uh, what do you expect to happen here? Where on the spaCy website did you try this?

I am really unsure what you want and what is wrong with the output you have here.

**Reply by ilcaa72** [reply]
the input was 
1.  This is an example. This is a numbered list.
2.  This is another example.  This is #2 of a numbered list

and the output was
1.
This is an example.
this is a numbered list.

2.
this is another example.
This is #2 of a numbered list.

do you not see why this is not correct?  is spacy not smart enough to know this is a numbered list starting a line and not to parse it? (obviously she doesn't), but more importantly, you think this is an acceptable output

**Reply by polm** [reply]
OK, I think I understand now. Is it correct that if your input is a numbered list like so...

```
1. This is an example. This is a numbered list.
2. This is another example. This is #2 of a numbered list
```

You want output like this:

```
["1. This is an example. This is a numbered list.", "2. This is another example. This is #2 of a numbered list"]
```

Is that right? 

I'm still not sure where you're running this code on the spaCy website, or exactly what code you're running, but your output looks like you're using `doc.sents`. That gives the sentences a doc is made up of. While it's not always perfect, in this case I think the output is normal - for example, this is three sentences, I'll put one on each line for clarity:

```
1.
This is an example.
This is a numbered list.
```

I can understand that you might want to treat a numbered list entry as a single sentence for your application. You can create a custom senter to handle that if you like. But by the normal definition of a sentence, the output here is correct.

If you were not trying to split a doc into sentences, then please explain what you are trying to do and I'd be glad to help you.

I will note that this part of the sentence splitting in the original looks wrong:

```
It is all about brainstorming with your teammates about what you want to build (e.g., Who are your users?
What will each user want from your product?).
```

But that seems to be easily explained by the unusual punctuation.

**Reply by ilcaa72** [reply]
i just assumed that i numbered list is not really considered a sentence and would not be parsed even though there technically are sentences in the numbered list. 

So sorry for the post, i wasnt aware i could write a custom senter. ill look into that as well.  thanks for the replies

**Reply by svlandeg** [reply]
No need to be sorry for the post :-) For future reference, we can better help you if you provide some sample code that shows us clearly what you're doing, though ;-)
